crates/runtime/src/http/v1/mod.rs

```diff
@@ -93,30 +94,50 @@ impl utoipa::IntoParams for Format {
 #[derive(Default, Debug, Serialize, Deserialize)]
 #[serde(rename_all = "lowercase")]
 /// The various formats that the Arrow data can be converted and returned from HTTP requests.
-pub enum ArrowFormat {
+pub enum DataFormat {
     #[default]
     Json,
     Csv,
     Plain,
+    VndJsonV1,
+}
+
+/// Represents additional metadata to produce a response, such as the SQL query used, etc.
+#[derive(Debug)]
+pub struct ResponseMetadata {
+    pub sql: Option<String>,
+}
+
+impl ResponseMetadata {
+    /// Creates an empty `ResponseMetadata`
+    pub fn empty() -> Self {
+        Self { sql: None }
+    }
+
+    pub fn with_sql(mut self, sql: impl Into<String>) -> Self {
+        self.sql = Some(sql.into());
+        self
+    }
 }
 
 /// Gets all possible media types from a `Accept` header.
 pub(crate) fn accept_header_types(accept: &TypedHeader<Accept>) -> Vec<String> {
     accept.0.media_types().map(ToString::to_string).collect()
 }
 
-impl ArrowFormat {
-    pub fn from_accept_header(accept: Option<&TypedHeader<Accept>>) -> ArrowFormat {
-        accept.map_or(ArrowFormat::default(), |header| {
+impl DataFormat {
+    pub fn from_accept_header(accept: Option<&TypedHeader<Accept>>) -> DataFormat {
+        accept.map_or(DataFormat::default(), |header| {
             accept_header_types(header)
                 .iter()
                 .find_map(|h| match h.as_str() {
-                    "application/json" => Some(ArrowFormat::Json),
-                    "text/csv" => Some(ArrowFormat::Csv),
-                    "text/plain" => Some(ArrowFormat::Plain),
+                    "application/json" => Some(DataFormat::Json),
+                    "application/vnd.spiceai.v1+json" => Some(DataFormat::VndJsonV1),
```

Comment:
Should this be specific for SQL? i.e. `application/vnd.spiceai.sql.v1+json`

That way we could have specific formats for other APIs.

Reply:
Changed to `application/vnd.spiceai.sql.v1+json`